packages/vite-plugin/src/index.ts

```diff
@@ -71,32 +99,38 @@ export function vanillaExtractPlugin({
         return null;
       }
 
-      const usedIndex = id.indexOf('?used');
-      const fixedId = usedIndex > 0 ? id.substring(0, usedIndex) : id;
-
-      if (ssr || useRuntime) {
+      if (ssr) {
+        const validId = id.substring(0, id.indexOf('?'));
         return addFileScope({
           source: code,
-          filePath: normalizePath(path.relative(packageInfo.dirname, fixedId)),
+          filePath: normalizePath(path.relative(packageInfo.dirname, validId)),
           packageInfo,
         }).source;
       }
 
       const { source, watchFiles } = await compile({
-        filePath: fixedId,
+        filePath: id,
```

Comment:
I think we should continue to strip the query if it has one, just to be sure there's no funny business.